pipeline/alignments/generate-shortlist.sh

```diff
@@ -63,9 +64,9 @@ fi
 rm "${dir}/corpus.spm.${TRG}"
 rm "${dir}/corpus.spm.${SRC}"
 rm "${output_dir}/corpus.aln"
-
+# TODO: check
 echo "### Shortlist pruning"
-"${MARIAN}/spm_export_vocab" --model="${vocab_path}" --output="${dir}/vocab.txt"
+"${MARIAN}/spm_export_vocab" --model="${vocab_trg}" --output="${dir}/vocab.txt"
```

Comment:
I spent some time trying to figure out how this works and using the target vocab looks correct here. The confusion is that like in the [original code](https://github.com/browsermt/students/blob/36cf85a94aa19edaf37afe97722e0e90fd5dc524/train-student/alignment/generate-alignment-and-shortlist.sh#L37) we do  `"${BIN}/extract_lex" "${dir}/corpus.spm.${TRG}" "${dir}/corpus.spm.${SRC}` so lex.s2t is in fact target to source. `prune_shortlist.py` expects this. I hope this works properly on the inference side and it takes into account that the target is the first column.

Reply:
Thinking through shortlisting a bit more, I think this is actually the correct thing as you are suggesting. The short listing behavior on the inference side doesn't need a short list of the source tokens, as it includes all of the source translation tokens in the shortlist so that they could potentially be copied over to the target. The most correct short list here would then be only the target vocab, as that's the only thing the inference engine would need to produce.

Reply:
I agree it's not well documented but this PR is pretty big and I think we'll be able to catch most of the bugs on evaluation if the shortlist is not correct. So yes, refactoring should be a follow-up here. I think I filed a bug about an end-to-end quality test in the past.